docs/SPEC.md

```diff
@@ -2067,34 +2067,43 @@ from(bucket:"telegraf/autogen")
 Group groups records based on their values for specific columns.
 It produces tables with new group keys based on the provided properties.
 
-Group has the following properties:
+Group accepts _exactly one_ of the following properties:
 
 *  `by` list of strings
     Group by these specific columns.
-    Cannot be used with `except`.
 *  `except` list of strings
     Group by all other columns except this list.
-    Cannot be used with `by`.
+*  `none` boolean
+    Group by nothing, so, group all records into a single group.
+*  `all` boolean
```

Comment:
Group all means that you are grouping by all columns and as such each unique row will be in its own group.

Reply:
Oh yes, got it. Indeed, it was pretty obvious 😄 . My doubt arose from the fact that I ran some queries with group all and the result was the same as no grouping, but maybe it is because the data was already grouped by every column by telegraf itself.

Let me be more precise: I issued this query on the `syslog` bucket generated by telegraf:

```
from(bucket: "telegraf")
    |> filter(fn: (r) => r._measurement == "syslog")
    |> range(start: -24h)
    |> group(all: true)
```

And I got (showing only the table header) this:

```
#datatype,string,long,dateTime:RFC3339,dateTime:RFC3339,dateTime:RFC3339,double,string,string,string,string
#group,false,false,true,true,false,false,true,true,true,true
#default,_result,,,,,,,,,
,result,table,_start,_stop,_time,_value,_field,_measurement,cpu,host
```

Is it true that "all" doesn't include `_time` and `_value`?

Reply:
Maybe, the best to tell it is like here: https://github.com/influxdata/flux/blob/master/functions/group_mode.go#L11

"GroupModeAll produces a separate table for each series."

Reply:
Also note that the work I did in influxdata/platform#1312 provides new implementations of these strategies when merging multiple remote results sets with additional unit tests.

Reply:
Thank you Stuart, I'll update the spec accordingly.

Reply:
Ok, I needed to run some tests to clarify some bits.

* What `group(none: true)` does is crystal clear;
* For what concerns `group(all: true)`, it performs a little different from what you say (at least from my results):

> the partition key == all the tag values of the series key

Sorry for being pedantic, but I would say that the partition key is composed of the values of `_field`, `_measurement`, and the tag values.  
If the "series key" is composed of the values of `_field` and `_measurement` values, then we are telling the same thing! 😄 

If it is so, I suggest to fix the "Concepts" section in the documentation, because it tells that a series is "collection of data that share a retention policy, measurement, and tag set" (https://docs.influxdata.com/influxdb/v1.7/concepts/key_concepts/#series). It becomes quite natural to think of the "series key" as measurement and tag set, but not field.

Reply:
It's worth noting in this discussion that `group(all: true)` is not implemented at all in the `Process` method and happens to behave the same as `group(none: true)` when evaluated in Flux.  I've been working on this issue to address it:
https://github.com/influxdata/flux/issues/322
So we will likely see differing behavior depending on whether grouping is evaluated in storage or in flux.  Note that if there's several consecutive `group` calls, we will only push the first one to storage, and evaluate the rest in Flux.

@stuartcarnie @affo The way `group(all: true)` is being described here I would expect it to be a no-op, or am I misunderstanding?  I think I agree with @nathanielc that `group(all: true)` should group by all the columns, including `_value` and `_time`.  Maybe we can't push down the grouping into storage for that case though.

When we're all available we should talk about what the right behavior should be for both Flux and storage.

Reply:
it looks to me that the `flux` syntax was overloaded to encode the storage-level RPC flags for GroupMode.  The semantics should come from there.  the issue with `group(all:true)`  is that it's only well-defined if we have a default set of influxDB storage columns.  instead of re-interpreting it in the `process` method, we should probably return an error instead, reporting that that mode is only compatible when pushed down to an influxdb storage request.  

I had never considered this before, or else I might have pushed for a different function signature.  Something like  group(auto: "none")   to get GroupModeNone  and group(auto: "influxdb-default") to get GroupModeAll.  basically, if there's automagic ways of doing grouping we should put them here.   This could also help us with an existing feature request to additively construct a group key, e.g.,  `group(auto: "concat-existing", by: ["newCol1", "newCol2"])`.  Though that's more of a hack than what we should be doing (schema introspection #27 ).

Reply:
Sorry for the delay.

> It's worth noting in this discussion that `group(all: true)` is not implemented at all in the `Process` method and happens to behave the same as `group(none: true)` when evaluated in Flux. I've been working on this issue to address it:
> #322
> So we will likely see differing behavior depending on whether grouping is evaluated in storage or in flux. Note that if there's several consecutive `group` calls, we will only push the first one to storage, and evaluate the rest in Flux.
> 

This sheds light on the suspicious behavior I detected in #322. When `group(all: true)` is the first group operation it is performed by storage, otherwise it is not performed because flux implements it this way. Great!

> @stuartcarnie @affo The way `group(all: true)` is being described here I would expect it to be a no-op, or am I misunderstanding? I think I agree with @nathanielc that `group(all: true)` should group by all the columns, including `_value` and `_time`. Maybe we can't push down the grouping into storage for that case though.
> 
> When we're all available we should talk about what the right behavior should be for both Flux and storage.

With a rewrite rule to use only the last grouping, it should be performed by storage and fix the problem, at least if it is possible to perform it on storage. If the user specifies transformations and then re-groups, the grouping is performed by flux and the problem araises.